### PR TITLE
Adding Error Handling in PVRGlobals & native-lib

### PR DESCRIPTION
**Pull request by thezofx**
For more descriptive logging, to resolve issues #49 and #43

### CODE REVIEW

File: code/mobile/mobile-common/PVRSockets.cpp [hunk 1]
```diff
@@ -355,19 +416,30 @@ void PVRStartReceiveStreams(uint16_t port) {
...
+            strThr->join();
+            delete strThr;
+            strThr = nullptr;
+        }
+    }
+    catch(exception e) {
+        PVR_DB("PVRSockets_PVRStopStreams:: Caught Exception: " + string(e.what()));
+    }
+    catch(...) {
+        PVR_DB("PVRSockets_PVRStopStreams:: Caught Exception: from Generic Handler");
```
Comment (ShootingKing-AM) [inline code/mobile/mobile-common/PVRSockets.cpp:443]: Can we remove Generic catch block ? because even if it catches this type of exception, we will not be getting any solid information except that "Caught Exception: from Generic Handler", not helpful in debugging. Rather, we can let the App crash and let Android `debuggerd` collect stacktrace, and we can request client to upload tombstones, which will be more useful rather than "Caught Exception: from Generic Handler" without any message.

File: code/mobile/mobile-common/Utils/RenderUtils.cpp [hunk 2]
```diff
@@ -105,7 +105,7 @@ void Renderer::initProg(vector<pair<GLuint, bool>> texs, std::string vs, string
 
 // This is the actuall Constructor
 Renderer::Renderer(vector<pair<GLuint, bool>> texs, string frag, float xStart, float xEnd)
-        : rtt(false), frmBuf(0), maxLod(0){
```
Comment (ShootingKing-AM) [inline code/mobile/mobile-common/Utils/RenderUtils.cpp:108]: Why this change ?

File: code/mobile/mobile-common/Utils/RenderUtils.cpp [hunk 3]
```diff
@@ -114,7 +114,7 @@ Renderer::Renderer(vector<pair<GLuint, bool>> texs, string frag, float xStart, f
 
 Renderer::Renderer(int width, int height, vector<pair<GLuint, bool>> texs, string frag, GLenum fmt,
                    GLint mag, GLuint rdrBuf, int max) : width(width), height(height),
-                                                        size(width * height * 4), rtt(true), maxLod(max) {
```
Comment (ShootingKing-AM) [inline code/mobile/mobile-common/Utils/RenderUtils.cpp:117]: Why this change ?

**Review by ShootingKing-AM — CHANGES_REQUESTED** [review]
Small suggestions, rest seems perfect 👍

File: code/common/src/Utils/ThreadUtils.h [hunk 4]
```diff
@@ -94,7 +94,7 @@ class TimeBomb
 			EndThread(thr);
 	}
 
-	TimeBomb(std::chrono::microseconds timer, std::function<void()> boomCb = [] {}) : tm(timer), cb(boomCb), loop(true) {}
```
[the remark this answers is not in the document]
Comment (thezofx) [reply code/common/src/Utils/ThreadUtils.h:97]: Change was done In order to fix compiler warnings but committed unknowingly.

File: code/mobile/mobile-common/Utils/RenderUtils.cpp (hunk 3 above)
Comment (thezofx) [reply code/mobile/mobile-common/Utils/RenderUtils.cpp:117]: Change was done In order to fix compiler warnings but committed unknowingly.

File: code/mobile/mobile-common/Utils/RenderUtils.cpp (hunk 2 above)
Comment (thezofx) [reply code/mobile/mobile-common/Utils/RenderUtils.cpp:108]: Change was done In order to fix compiler warnings but committed unknowingly.

File: code/mobile/mobile-common/PVRRenderer.cpp [hunk 5]
```diff
@@ -34,9 +34,19 @@ namespace {
...
+            for (int i = 0; i < 4; ++i) {
+                for (int j = 0; j < 4; ++j) {
+                    eMat(i, j) = gvrMat.m[i][j];
+                }
+            }
+        }
+        catch(exception e) {
+            PVR_DB("PVRRenderer_gvrToEigenMat:: Caught Exception: " + string(e.what()));
+        }
+        catch(...) {
```
Comment (ShootingKing-AM) [inline code/mobile/mobile-common/PVRRenderer.cpp:47]: Still many Generic Catchs are implemented, and many exception messages are PVR_DB instead of PVR_DB_I

**Comment by ShootingKing-AM** [comment]
After you make requested changes, clikc "Resolve converstation". Did you build and test these, i am getting a "black screen" if i build both server and android with these changes.

**Review by ShootingKing-AM — APPROVED** [review]
